core/maximal_repeats.py

```diff
@@ -55,43 +65,34 @@ def is_empty(self) -> bool:
 
 def process_leaves(lcp_interval: Interval, suf_tab: list[int], bwt_table: list[str]) -> None:
     lb, rb, lcp = lcp_interval.lb, lcp_interval.rb, lcp_interval.lcp_value
-    pos_sets: PositionSets = {}
-
-    for alphabet in alphabet_set:
-        for i in range(lb, rb + 1):
-            if bwt_table[i] == alphabet:
-                if alphabet in pos_sets:
-                    curr_child_alphabet_itr = pos_sets[alphabet]
-
-                    while curr_child_alphabet_itr.next is not None:
-                        curr_child_alphabet_itr = curr_child_alphabet_itr.next
-
-                    curr_child_alphabet_itr.next = ListNode(suf_tab[i])
+    parent_pos_sets: PositionSets = {}
+    last_leaf_pos_set: PositionSets = {}
 
-                else:
-                    pos_sets[alphabet] = ListNode(suf_tab[i])
+    for i in range(lb, rb + 1):
+        alphabet: str = bwt_table[i]
+        curr_child_pos_set: PositionSets = {alphabet: ListNode(suf_tab[i])}
 
-    # process the leaves
-    for a in pos_sets:
-        for b in pos_sets:
-            if a != b:
-                a_itr = pos_sets[a]
-                b_itr_head = pos_sets[b]
-
-                while a_itr is not None:
-                    b_itr = b_itr_head
-                    p = a_itr.data
-
-                    while b_itr is not None:
-                        p_prime = b_itr.data
+        if len(last_leaf_pos_set.keys()) == 0:
```

Comment:
Is this for testing if `last_leaf_pos_set` is empty? Then it is a peculiar way to do it. If you use a set or dict or list as a truth value it evaluates to false if it is empty, so `if not last_leaf_pos_set` should do the same thing. (Unless there is something odd with `last_leaf_pos_set`).